### DISCUSSION TITLE
Standard branches and commits naming?

### DISCUSSION THREAD

**Discussion by Sharaf-Mansour** [hassanhabib/The-Standard#225] [Q&A] [answered]
It should be clear how we name our branches and commits according to **the standard**
Like a dedicated section for **version control** to show the true beauty of the standard 
(Telling the whole story of the project)

**Comment by cjdutoit** [comment]
### Branches
For branches we can apply the following naming convention:  users\\`%username%` \\`%category%`-`%entity%`-`%action%`
`%username%` =>  your github username
`%category%` => the category that you are working on 
`%entity%` => the entity that your service / broker uses
`%action%` => the action

e.g. 
`users\cjdutoit\brokers-students-add`
`users\cjdutoit\foundations-students-modify`

For a complete list of categories see [https://gitfyle.com/about](https://gitfyle.com/about)

### Commits
For commits we can use the following naming convention: `%category%`: `%entity%` - `%action%`
`%category%` => the category that you are working on  (ALL CAPS)
`%entity%` => the entity that your service / broker uses (Pascal Case)
`%action%` => the action (Pascal Case)

e.g. 
`BROKERS: Student Broker - Add`
`FOUNDATIONS: Student Service - Modify`

If your commit is related to an issue, you can link a commit to a GitHub Issue by putting the issue number with the `#` character in the Comment text box. For example, `#111`.

To close a GitHub issue, put the keyword `Closes` and the issue number with the `#` character in the Comment text box. 
For example, `Closes #111`.

**Reply by Sharaf-Mansour** [reply]
Amazing! @cjdutoit Thanks so much!

I hope that we add this to its own section in **the standard** and have @hassanhabib Explain what each category e.g.  ```INFRA, PROVISION, etc.``` Means and when to use which. 

Would be very helpful for students and newcomers to learn and enjoy coding.

**Comment by cjdutoit** [comment, marked as answer]
Documentation can be found here:  https://github.com/hassanhabib/The-Standard-Team/blob/main/4%20Practices/4%20Practices.md
